agent/structs/structs.go

```diff
@@ -2391,3 +2408,11 @@ func (r *KeyringResponses) Add(v interface{}) {
 func (r *KeyringResponses) New() interface{} {
 	return new(KeyringResponses)
 }
+
+type UpstreamDownstream struct {
+	Upstream   ServiceName
+	Downstream ServiceName
+	Refs       map[string]bool
```

Comment:
What is `Refs` used for?

Reply:
Tracks the registrations that relate to this UpstreamDownstream. Added a comment.

I had a RefCount at first but now I don't remember why that didn't work.